Assets/Mirror/Runtime/NetworkIdentity.cs

```diff
@@ -835,58 +838,44 @@ public void RebuildObservers(bool initialize)
             }
 
             // if no component implemented OnRebuildObservers, then add all
-            // connections.
+            // connections that are ready
             if (!result)
             {
-                if (initialize)
+                foreach (NetworkConnection conn in NetworkServer.connections.Values)
                 {
-                    foreach (NetworkConnection conn in NetworkServer.connections.Values)
-                    {
-                        if (conn.isReady)
-                            AddObserver(conn);
-                    }
-
-                    if (NetworkServer.localConnection != null && NetworkServer.localConnection.isReady)
-                    {
-                        AddObserver(NetworkServer.localConnection);
-                    }
+                    if (conn.isReady)
+                        newObservers.Add(conn);
                 }
-                return;
             }
 
-            // apply changes from rebuild
+            // our list is fully built,  
+            // spawn this object for every new observer
             foreach (NetworkConnection conn in newObservers)
             {
-                if (conn == null)
-                {
-                    continue;
-                }
-
-                if (!conn.isReady)
+                if (conn != null && conn.isReady)
                 {
-                    if (LogFilter.Debug) Debug.Log("Observer is not ready for " + gameObject + " " + conn);
-                    continue;
-                }
-
-                if (initialize || !oldObservers.Contains(conn))
-                {
-                    // new observer
-                    conn.AddToVisList(this);
```

Comment:
why is conn.AddToVisList gone in your PR?

Reply:
because AddObserver calls it already